resources/webgui_minimal/index.html

```diff
@@ -7,34 +7,104 @@
     <script src="script/osc-browser.min.js"></script>
     <script>
       var oscPort = new osc.WebSocketPort({
-          url: "ws://localhost:8080/ws",
+          url: "ws://localhost:7080/ws",
           metadata: true
       });
       oscPort.open();
 
-      function send(address) {
+      function send(address, value) {
+          oscPort.send({
+              address: address,
+	      args: [
+		  {
+		      type: "s",
+		      value: value
+		  }
+	      ]
+          });
+      };
+
+      function sendResponse(address) {
+          oscPort.send({
+              address: address,
+	      args: [
+		  {
+		      type: "s",
+		      value: "connect"
+		  },
+		  {
+		      type: "s",
+		      value: "localhost"
+		  },
+		  {
+			  type: "i",
+			  value: 9336
+		  }
+		  ]
+          });
+      };
+
+      function sendNoArgs(address) {
           oscPort.send({
               address: address
           });
       };
 
       oscPort.on("ready", function () {
-          send("/ping");
+          oscPort.send({ address: "/ping" });
+      });
+
+      oscPort.on("message", function (oscMsg) {
+		console.log("OSC message received: ", oscMsg);
+
+		path = oscMsg.address.substring(1).split("/");
+
+		if (path[0] == "app" && path[2] == "state") {
+			// Show status info.
+			document.getElementById(path[1] + "_status").innerHTML = oscMsg.args[1].value;
```

Comment:
what happens if this HTML element doesn't exist?

Reply:
Well yes, that's entirely non-production code and just for testing, so I would not invest much here…